portal/worker/src/walrus-sites-sw.ts

```diff
@@ -50,8 +54,40 @@ self.addEventListener("fetch", async (event) => {
     console.log("Parsed URL: ", parsedUrl);
 
     if (requestDomain == portalDomain && parsedUrl && parsedUrl.subdomain) {
-        console.log("fetching from the service worker");
-        event.respondWith(resolveAndFetchPage(parsedUrl));
+        event.respondWith((async () => {
+            // Clean the cache of expired entries
+            await cleanExpiredCache();
+            if (!('caches' in self)) {
+                // When not being in a secure context, the Cache API is not available.
+                console.warn('Cache API not available');
+                return await resolveAndFetchPage(parsedUrl);
+            }
+
+            const cache = await caches.open(cacheName);
+            const cachedResponse = await cache.match(urlString);
+            let isCacheSameAsNetwork: boolean;
+            try {
+                if (cachedResponse) {
+                    isCacheSameAsNetwork = await checkCachedVersionMatchesOnChain(cachedResponse)
```

Comment:
Note for the future: we may want to optimize this (as discussed) and not always check if there is an updated version -- which costs another RTT to the full node.

The other problem of caching based on the object version is that the object version may change, without the object itself changing at all, triggering a re-fetch.

Reply:
Created an issue for this: https://github.com/MystenLabs/walrus-sites/issues/194